src/adaptors/adaptor_common.h

```diff
@@ -0,0 +1,114 @@
+#ifndef __adaptor_common_h__
+#define __adaptor_common_h__ 1
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include "entity.h"
+
+#include "qpid/dispatch/dispatch.h"
+#include "qpid/dispatch/threading.h"
+#include "qpid/dispatch/atomic.h"
+#include "qpid/dispatch/log.h"
+#include "qpid/dispatch/alloc_pool.h"
+
+#include <proton/tls.h>
+
+typedef enum {
+    VERSION_HTTP1,
+    VERSION_HTTP2,
+} qd_http_version_t;
+
+typedef enum {
+    QD_AGGREGATION_NONE,
+    QD_AGGREGATION_JSON,
+    QD_AGGREGATION_MULTIPART
+} qd_http_aggregation_t;
+
+// Currently we only have TCP and HTTP (HTTP/1 and HTTP/2) adaptors
+typedef enum {
+    TCP,
+    HTTP
+} qd_adaptor_type_t;
+
+typedef struct qd_adaptor_config_t qd_adaptor_config_t;
+
+struct qd_adaptor_config_t
+{
+    char              *name;
+    char              *host;
+    char              *port;
+    char              *address;
+    char              *site_id;
+    char              *host_port;
+    qd_dispatch_t     *qpid_dispatch;
+
+    //TLS related info
+    char              *ssl_profile_name;
+    bool               require_tls;
+    bool               authenticate_peer;
+    bool               verify_host_name;
+
+    union {
+       struct {
+           qd_http_version_t      version;
+           bool                   event_channel;
+           qd_http_aggregation_t  aggregation;
+           char                  *host_override;
+       } http;
+
+       struct {
+        // run time statistics updated by connections
+        sys_mutex_t  *stats_lock;
+        uint64_t      connections_opened;
+        uint64_t      connections_closed;
+        uint64_t      bytes_in;
+        uint64_t      bytes_out;
+        sys_atomic_t  ref_count;
```

Comment:
What is "ref_count" counting?  If ref_count is counting references to the entire qd_adaptor_config_t then this ref_count should be moved out of the union and made a top level field.